client/lib/form-base/reducer.js

```diff
@@ -0,0 +1,45 @@
+import { SET_FORM_DATA_VALUE, SET_FORM_META_PROPERTY } from './actions';
+
+export const initialState = {
+	meta: {
+		error: false,
+		fieldsStatus: false,
+		isSaving: false,
+		pristine: true,
+	},
+	data: {
+	},
+};
+
+const reducers = {};
+
+reducers[ SET_FORM_DATA_VALUE ] = ( state, action ) => {
+	const data = Object.assign( {}, state.data, { [ action.key ]: action.value } );
```

Comment:
For an alternative to this `Object.assign` syntax, see: http://redux.js.org/docs/recipes/UsingObjectSpreadOperator.html
Example in our codebase: https://github.com/Automattic/woocommerce-connect-client/pull/491/files#diff-a145a70f9fe0565fb7752d16400d20cbR25
Of course, to use one syntax or another is entirely personal :)